safe_webrtc_example/app/safe_comm.js

```diff
@@ -0,0 +1,624 @@
+        keysHandle[CONST.CRYPTO_KEYS.SEC_ENC_KEY] = secEncKey;
+        const secEncKeyRaw = await window.safeCryptoSecEncKey.getRaw(secEncKey);
+        const encSecEncKey = await window.safeMutableData.encryptValue(this.channelMD, secEncKeyRaw.buffer);
+        const secEncKeyArr = utils.bufToArr(encSecEncKey);
+
+        const entries = {};
+        entries[CONST.CRYPTO_KEYS.PUB_ENC_KEY] = pubEncKeyArr;
+        entries[CONST.CRYPTO_KEYS.SEC_ENC_KEY] = secEncKeyArr;
+        await window.safeMutableData.quickSetup(this.channelMD, entries, 'WebRTC Channel', `WebRTC channel for ${hostName}`);
+
+        // create a new permission set
+        const permSet = [CONST.PERMISSIONS.READ, CONST.PERMISSIONS.INSERT];
+        await window.safeMutableData.setUserPermissions(this.channelMD, null, permSet, 1);
+
+        const channelSerialData = await window.safeMutableData.serialise(this.channelMD);
+        const entriesHandle = await window.safeMutableData.getEntries(this.serviceCntr);
+        const mutationHandle = await window.safeMutableDataEntries.mutate(entriesHandle);
+        const encryptedKey = await window.safeMutableData.encryptKey(this.serviceCntr, CONST.MD_KEY);
+        await window.safeMutableDataMutation.insert(mutationHandle, encryptedKey, channelSerialData);
+        await window.safeMutableData.applyEntriesMutation(this.serviceCntr, mutationHandle);
+        window.safeMutableDataMutation.free(mutationHandle);
+        window.safeMutableDataEntries.free(entriesHandle);
+        this._setKeys(keysHandle);
+        resolve(true);
+      } catch (err) {
+        utils.putLog('Create channel error', err);
+        reject(err);
+      }
+    });
+  }
+
+  _getPublicNamesCntr() {
+    utils.putLog('Get public names container');
+    return new Promise(async (resolve, reject) => {
+      try {
+        if (!this.app) {
+          return reject(new Error('Unauthorised'));
+        }
+        const hasAccess = await window.safeApp.canAccessContainer(this.app, '_publicNames', APP.containers._publicNames);
+        if (!hasAccess) {
+          return reject(new Error('No publicNames container access found'));
+        }
+        this.pubNameCntr = await window.safeApp.getContainer(this.app, '_publicNames');
+        resolve(true);
+      } catch (err) {
+        utils.putLog('Get public names container error', err);
+        reject(err);
+      }
+    });
+  }
+
+  _getSelectedPublicName() {
+    utils.putLog('Select public name');
+    return new Promise(async (resolve, reject) => {
+      try {
+        if (!this.app) {
+          return reject(new Error('Unauthorised'));
+        }
+        const ownCntr = await window.safeApp.getOwnContainer(this.app);
+        this.selectedPubName = await window.safeMutableData.get(ownCntr, CONST.SELECTED_PUB_NAME_KEY);
+        resolve(true);
+      } catch (err) {
+        // if (err.code !== CONST.ERR_CODE.NO_SUCH_ENTRY) {
+        if (err.message !== ERROR_MSG.ENTRY_NOT_FOUND) {
```

Comment:
I'm assuming error codes are not available from the DOM API? If so, I would add a comment (TODO/FIXME) here to state that we should use error codes whenever they become available.

Reply:
I've added some comments for things I'd change, but I thing the most important thing would be that this PR is breaking the CI, apparently just due to a missing `test` script?